code/test/sixsq/nuvla/server/resources/voucher_lifecycle_test.clj

```diff
@@ -0,0 +1,227 @@
+        session-user (header session-anon authn-info-header "user/abcdef01-abcd-abcd-abcd-abcdef012346 group/nuvla-user group/nuvla-anon")
+
+        valid-voucher-admin {:name               "my-voucher"
+                             :description        "my-voucher description"
+
+                             :owner              "user/abcdef01-abcd-abcd-abcd-abcdef012345"
+                             :amount             50.0
+                             :currency           "EUR"
+                             :code               "vH72Hks209"
+                             :state              "NEW"
+                             :target-audience    "scientists@university.com"
+
+                             :acl                valid-acl-admin
+                             }
+
+        valid-voucher-user (assoc valid-voucher-admin :acl valid-acl-user)]
+
+    ;; admin/user query succeeds but is empty
+    (doseq [session [session-admin session-user]]
+      (-> session
+          (request base-uri)
+          (ltu/body->edn)
+          (ltu/is-status 200)
+          (ltu/is-count zero?)
+          (ltu/is-operation-present "add")
+          (ltu/is-operation-absent "delete")
+          (ltu/is-operation-absent "edit")))
+
+    ;; anon query fails
+    (-> session-anon
+        (request base-uri)
+        (ltu/body->edn)
+        (ltu/is-status 403))
+
+    ;; anon create must fail
+    (-> session-anon
+        (request base-uri
+                 :request-method :post
+                 :body (json/write-str valid-voucher-admin))
+        (ltu/body->edn)
+        (ltu/is-status 403))
+
+    ;; check voucher creation
+    (let [admin-uri (-> session-admin
+                        (request base-uri
+                                 :request-method :post
+                                 :body (json/write-str valid-voucher-admin))
+                        (ltu/body->edn)
+                        (ltu/is-status 201)
+                        (ltu/location))
+
+          admin-abs-uri (str p/service-context admin-uri)
+
+          user-uri (-> session-user
+                       (request base-uri
+                                :request-method :post
+                                :body (json/write-str valid-voucher-user))
+                       (ltu/body->edn)
+                       (ltu/is-status 201)
+                       (ltu/location))
+
+          user-abs-uri (str p/service-context user-uri)]
+
+      ;; admin should see 2 voucher resources
+      (-> session-admin
+          (request base-uri)
+          (ltu/body->edn)
+          (ltu/is-status 200)
+          (ltu/is-resource-uri t/collection-type)
+          (ltu/is-count 2))
+
+      ;; user also sees 2 cause of acls
+      (-> session-user
+          (request base-uri)
+          (ltu/body->edn)
+          (ltu/is-status 200)
+          (ltu/is-resource-uri t/collection-type)
+          (ltu/is-count 2))
```

Comment:
@loomis this is where the test is failing. this is somehow related with the `view-data` acl.